packages/zowe-explorer-api/src/fs/BaseProvider.ts

```diff
@@ -417,6 +429,24 @@ export class BaseProvider {
         return entry;
     }
 
+    protected _handleError(err: Error, opts?: HandleErrorOpts): void {
+        ErrorCorrelator.getInstance()
+            .displayError(opts?.apiType ?? ZoweExplorerApiType.All, err, {
+                additionalContext: opts?.additionalContext,
+                allowRetry: opts?.retry?.fn != null,
+                profileType: opts?.profileType ?? "any",
+                templateArgs: opts?.templateArgs,
+            })
+            .then(async ({ userResponse }) => {
+                if (userResponse === "Retry" && opts?.retry?.fn != null) {
+                    await opts.retry.fn(...(opts?.retry.args ?? []));
+                }
+            })
+            .catch(() => {
+                throw err;
```

Comment:
Good call to add a log message to this catch block - we'll still need to throw the original error so that VS Code's file system handling is aware of the error that occurred in the filesystem.

Reply:
I now recall why I did not add an error message here - we do not expose the `ZoweLogger` to Zowe Explorer API. I still think we should add a log to this block, but not sure the best way to add it without adding a good chunk of code to expose it to the API (scope creep & more for reviewers to look over). 

Would you be opposed to adding a `console.log` call? I don't expect errors to be thrown from the correlator, but I added this catch block so that ESLint was satisfied with this use of a promise.

Reply:
console.error might owkr for now 😋 
but I think it's be good to figure out how to do proper logging from the ZE APIs 😅 

Thanks for considering this 🙏

Reply:
Created: 
- https://github.com/zowe/zowe-explorer-vscode/issues/3285